monai/transforms/spatial/array.py

```diff
@@ -1206,28 +1251,40 @@ class Rand2DElastic(Randomizable, Transform):
 
     def __init__(
         self,
-        spacing,
-        magnitude_range,
+        spacing: Sequence[float],
+        magnitude_range: Tuple[float, float],
         prob: float = 0.1,
-        rotate_range=None,
-        shear_range=None,
-        translate_range=None,
-        scale_range=None,
-        spatial_size=None,
+        rotate_range: Optional[Union[Sequence[float], float]] = None,
+        shear_range: Optional[Union[Sequence[float], float]] = None,
+        translate_range: Optional[Union[Sequence[float], float]] = None,
+        scale_range: Optional[Union[Sequence[float], float]] = None,
+        spatial_size: Optional[Union[Sequence[int], int]] = None,
         mode: Union[GridSampleMode, str] = GridSampleMode.BILINEAR,
         padding_mode: Union[GridSamplePadMode, str] = GridSamplePadMode.REFLECTION,
         as_tensor_output: bool = False,
         device: Optional[torch.device] = None,
     ):
         """
         Args:
-            spacing (2 ints): distance in between the control points.
-            magnitude_range (2 ints): the random offsets will be generated from
-                ``uniform[magnitude[0], magnitude[1])``.
+            spacing : distance in between the control points.
+            magnitude_range: the random offsets will be generated from ``uniform[magnitude[0], magnitude[1])``.
             prob: probability of returning a randomized affine grid.
                 defaults to 0.1, with 10% chance returns a randomized grid,
                 otherwise returns a ``spatial_size`` centered area extracted from the input image.
-            spatial_size (2 ints): specifying output image spatial size [h, w].
+            rotate_range: angle range in radians. rotate_range[0] with be used to generate the 1st rotation
+                parameter from `uniform[-rotate_range[0], rotate_range[0])`.
+            shear_range: shear_range[0] with be used to generate the 1st shearing parameter from
+                `uniform[-shear_range[0], shear_range[0])`. Similarly, `shear_range[1]` controls
+                the range of the uniform distribution used to generate the 2nd parameter.
+            translate_range : translate_range[0] with be used to generate the 1st shift parameter from
+                `uniform[-translate_range[0], translate_range[0])`. Similarly, `translate_range[1]` controls
+                the range of the uniform distribution used to generate the 2nd parameter.
+            scale_range: scaling_range[0] with be used to generate the 1st scaling factor from
+                `uniform[-scale_range[0], scale_range[0]) + 1.0`. Similarly, `scale_range[1]` controls
+                the range of the uniform distribution used to generate the 2nd parameter.
+            spatial_size: specifying output image spatial size [h, w].
+                if `spatial_size` and `self.spatial_size` are not defined, or smaller than 1,
```

Comment:
If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.